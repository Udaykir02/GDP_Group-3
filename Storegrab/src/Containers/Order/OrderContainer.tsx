import React, { useContext, useEffect, useState } from 'react';
import { FlatList, Modal, TouchableOpacity, View } from 'react-native';
import { Text } from 'react-native-paper';

import { useAppTheme } from '../../styles/theme/theme';
import AppPageWrapper from '../../shared/AppPageWrapper';
import LocationContainer from '../../Components/Location/LocationContainer';
import DeliveryCard from '../../Components/DeliveryCard';
import { useDispatch, useSelector } from 'react-redux';
import { getOrders, placeOrders } from '../../../actions/userActions';
import createToBeImplementedStyle from "../Order/OrderContainerStyle";
import { getVendorRequestByID } from '../../../actions/vendorActions';
import LoadingSpinner from '../../Components/shared/LoadingSpinner';

const OrderContainer: React.FC = () => {
  const [isModalVisible, setIsModalVisible] = useState(true);
  const styles = createToBeImplementedStyle();
  const dispatch = useDispatch();
  const { user, vendorAdmin, token } = useSelector((state: any) => state.auth)
  const { orders, loading } = useSelector((state: any) => state.order)
  const { selectedVendor } = useSelector((state: any) => state.vendor)
  useEffect(() => {
    if (vendorAdmin) {
      if (selectedVendor && selectedVendor?.vendorId)
        dispatch(getVendorRequestByID(selectedVendor?.vendorId, token))
    } else {
      console.log('user', user?.userId ? user.userId : '')
      dispatch(getOrders(user?.userId ? user.userId : '', token))
    }
  }, [vendorAdmin])

  useEffect(() => {
    console.log('orders', orders)
  }, [orders])

  const { colors } = useAppTheme();
  if (loading)
    return (<AppPageWrapper>
      <LoadingSpinner />
    </AppPageWrapper>)
  return (
    <AppPageWrapper>
      <FlatList
        data={orders}
        // ListHeaderComponent={renderHeader}
        renderItem={({ item, index }: any) => (
          <DeliveryCard order={item} />
        )}
        keyExtractor={(item, index) => index.toString()}
        extraData={orders}
        onEndReachedThreshold={0.1}
        //   ListFooterComponent={(moreproducts) ? <View row space="evenly" style={{ paddingVertical: 16 }}><View flex center><ActivityIndicator size="large" /></View></View> : null}
        refreshing={loading}
        // onEndReached={() => fetchMore()}
        style={styles.articles}
        showsVerticalScrollIndicator={false}
      />
    </AppPageWrapper>
  );
};

export default OrderContainer;
