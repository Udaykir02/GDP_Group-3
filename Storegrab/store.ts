import { configureStore } from '@reduxjs/toolkit';
import createSagaMiddleware from 'redux-saga';
import rootReducers from './reducers'; // Assuming rootReducers is defined in a separate file
import rootSaga from './sagas/rootSaga'; // Assuming rootSaga is defined in a separate file

// Create saga middleware instance
const sagaMiddleware = createSagaMiddleware();

// Create the Redux store with middleware and reducers
const store = configureStore({
  reducer: rootReducers,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(sagaMiddleware),
});

// Run the root saga
sagaMiddleware.run(rootSaga);

export default store;
