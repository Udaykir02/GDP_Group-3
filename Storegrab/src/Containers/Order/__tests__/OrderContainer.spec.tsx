import { Provider } from 'react-redux'
import store from '../../../../store'
import render, {
    RenderResult,
} from '@testing-library/react-native/build/render'
import OrderContainer from '../OrderContainer'
import { NavigationContainer } from '@react-navigation/native'
jest.mock('@react-navigation/native', () => {
    return {
      ...jest.requireActual('@react-navigation/native'),
      useNavigation: () => ({
        navigate: jest.fn(),
      }),
    };
  });
const renderStartupContainerPage = () => {
    return render(
        <Provider store={store}>
            <NavigationContainer>
                <OrderContainer />
            </NavigationContainer>
        </Provider>,
    )
}
let page: RenderResult
describe('VendorSubscriptionPage', () => {
    beforeEach(() => {
        jest.useFakeTimers();
        page = renderStartupContainerPage()
    })
    it('should render correctly', function () {
        expect(page.toJSON()).toMatchSnapshot()
    })

});