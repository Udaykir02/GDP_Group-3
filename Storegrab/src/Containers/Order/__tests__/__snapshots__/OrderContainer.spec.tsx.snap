// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VendorSubscriptionPage should render correctly 1`] = `
<View
  style={
    {
      "backgroundColor": undefined,
      "flexDirection": "column",
      "padding": 16,
      "width": "100%",
    }
  }
>
  <RCTScrollView
    data={[]}
    extraData={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={0.1}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshing={false}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={[]}
    style={
      {
        "paddingHorizontal": 2,
        "width": "auto",
      }
    }
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
</View>
`;
